### PR TITLE
Created an UpScale3D Layer

### PR DESCRIPTION
**Pull request by rohitghosh**
Upscale 3Dlayer created in lines with Upscale2D layer for using in recently published volumetric nets such as 3D UNets . Hope  it is fine !

### CODE REVIEW

**Comment by benanne** [comment]
Looks good to me! It would be great if someone who has used `Upscale2DLayer` before could take a look as well (because I haven't).

File: lasagne/layers/pool.py [hunk 1]
```diff
@@ -467,6 +468,59 @@ def get_output_for(self, input, **kwargs):
...
+        superclass.
+    """
+
+    def __init__(self, incoming, scale_factor, **kwargs):
+        super(Upscale3DLayer, self).__init__(incoming, **kwargs)
+
+        self.scale_factor = as_tuple(scale_factor, 3)
+
+        if self.scale_factor[0] < 1 or self.scale_factor[1] < 1 or \
+           self.scale_factor[2] < 1:
```
Comment (f0k) [inline lasagne/layers/pool.py]: Maybe nicer: `if min(self.scale_factor) < 1:`

**Comment by f0k** [comment]
Looks good to me. Haven't used Upscale2DLayer either, but it's a straightforward extension of Upscale2DLayer (could be refactored into an UpscaleLayer with three subclasses in the future).

Before we merge, could you please squash all your changes into a single commit? It should be something like:
```
git reset --soft dbd176a
git commit --amend --no-edit
git push --force
```
Thanks!

**Comment by bthyreau** [comment]
Looks good to me too. I use the exact same code, as a custom layer, in production for some time.

**Comment by f0k** [comment]
Thanks for checking, @bthyreau!

@rohitghosh, could you please squash the commits so we can merge (commands given above)? Thank you!

**Comment by rohitghosh** [comment]
@f0k : I tried running the above commands to squash the commits.  However I get stuck in the part 
`git push --force`
I get the error : Permission to Lasagne/Lasagne.git denied to rohitghosh. I am a bit new to Github, could you please help me out there ?

**Comment by rohitghosh** [comment]
@f0k : Done. Hope it's okay now !

**Comment by f0k** [comment]
Perfect, merging! Thank you for your contribution.

**Pull request merged**